src/shared_modules/sync_protocol/src/persistent_queue_storage.cpp

```diff
@@ -391,3 +391,62 @@ void PersistentQueueStorage::deleteDatabase()
         throw;
     }
 }
+
+void PersistentQueueStorage::addDataContextColumn()
+{
+    try
+    {
+        // Add is_data_context column if it doesn't exist (for VD sync databases)
+        const std::string alterQuery =
+            "ALTER TABLE persistent_queue ADD COLUMN is_data_context INTEGER NOT NULL DEFAULT 0;";
+
+        m_connection.execute(alterQuery);
+        m_logger(LOG_INFO, "PersistentQueueStorage: Added is_data_context column for DataContext support");
+    }
+    catch (const std::exception& ex)
+    {
+        // Column might already exist - log as debug instead of error
+        m_logger(LOG_DEBUG, std::string("PersistentQueueStorage: is_data_context column may already exist: ") + ex.what());
+    }
+}
+
+std::vector<PersistedData> PersistentQueueStorage::getAllEvents()
+{
+    std::vector<PersistedData> result;
+
+    try
+    {
+        // This is EXACTLY the same query as fetchAndMarkForSync() but WITHOUT the UPDATE
+        std::string selectQuery =
+            "SELECT rowid, id, idx, data, operation, "
+            "COALESCE(is_data_context, 0) as is_data_context "
```

Comment:
COALESCE is no longer needed here since the is_data_context column now has a NOT NULL